src/testrender/cuda/rend_lib.cu

```diff
@@ -225,7 +225,10 @@ extern "C" {
     __device__
     void osl_printf (void* sg_, char* fmt_str, void* args)
     {
-        printf (fmt_str, args);
+        // FIXME: This is to limit printing to one Cuda thread
+        //        Mainly to keep the testsuite happy.
+        if (launch_index.x == 0 && launch_index.y == 0)
+            vprintf(fmt_str, (const char*) args);
```

Comment:
Optix has its own printf call that I believe has extra hooks for you to control externally which launch id its active on. Would it be possible/desireable to use that one here?

Reply:
Cuda's printf get's compiled to vprintf.
https://stackoverflow.com/questions/22027955/using-ellipsis-in-cuda-device-function